src/Sylius/Bundle/ApiBundle/Validator/Constraints/CorrectOrderAddressValidator.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Validator\Constraints;
+
+use Sylius\Bundle\ApiBundle\Command\Checkout\AddressOrder;
+use Sylius\Component\Core\Model\AddressInterface;
+use Symfony\Component\Intl\Countries;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+final class CorrectOrderAddressValidator extends ConstraintValidator
+{
+    public function validate($value, Constraint $constraint)
+    {
+        /** @var AddressOrder $value */
+        Assert::isInstanceOf($value, AddressOrder::class);
+
+        /** @var CorrectOrderAddress $constraint */
+        Assert::isInstanceOf($constraint, CorrectOrderAddress::class);
+
+        $this->validateAddress($value->billingAddress, $constraint);
+        $this->validateAddress($value->shippingAddress, $constraint);
+    }
+
+    private function validateAddress(?AddressInterface $address, CorrectOrderAddress $constraint): void
+    {
+        if ($address === null) {
+            return;
+        }
+
+        /** @var string|null $countryCode */
+        $countryCode = $address->getCountryCode();
+
+        if ($countryCode === null) {
```

Comment:
I'm wondering, isn't it the case for the other validator 😄 We would then have two constraints e.g. `CountrySpecified` and `CountryExisting`? Seems as a better separation of responsibilities

Reply:
Its possible, but IMO it is just `addressValidator`. I can extract it to other services, but is it too much code to simple address validation? :D And we can add here next points for other address fields.